From d2a6c379e85bacc89a7a8468f7e8447c8df15785 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@collabora.com>
Date: Fri, 15 Mar 2024 13:49:47 +0000
Subject: [PATCH] girparser: Don't assume sizeof(size_t) == sizeof(void *)

We don't actually need to use the results of configure-time checks here:
sizeof is a perfectly reasonable integer constant expression, so we can
use that directly.

Helps: https://gitlab.gnome.org/GNOME/glib/-/issues/2842
Signed-off-by: Simon McVittie <smcv@collabora.com>
---
 girepository/girparser.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/girepository/girparser.c b/girepository/girparser.c
index 647cf2498d..b5d8fc7108 100644
--- a/girepository/girparser.c
+++ b/girepository/girparser.c
@@ -459,8 +459,8 @@ static IntegerAliasInfo integer_aliases[] = {
   { "gulong",   SIZEOF_LONG,      0 },
   { "gssize",   GLIB_SIZEOF_SIZE_T,    1 },
   { "gsize",    GLIB_SIZEOF_SIZE_T,    0 },
-  { "gintptr",  GLIB_SIZEOF_SIZE_T,    1 },
-  { "guintptr", GLIB_SIZEOF_SIZE_T,    0 },
+  { "gintptr",  sizeof (gintptr),      1 },
+  { "guintptr", sizeof (guintptr),     0 },
 };
 
 typedef struct {
-- 
GitLab

